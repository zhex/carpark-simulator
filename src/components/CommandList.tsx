import { ICmd } from 'models/interfaces/ICmd';
import * as React from 'react';
import styled from 'styled-components';

export interface ICommandListProps {
    commands: ICmd[];
    className?: string;
    currentIdx?: number;
}

const ListItem = styled.li`
    padding: 10px;
    border-bottom: 1px solid #ccc;
    &:last-of-type {
        border-bottom: none;
    }
`;

const ListItemActive = ListItem.extend`
    background: lightgreen;
`;

const CommandList: React.SFC<ICommandListProps> = props => (
    <ul className={props.className}>
        {props.commands.map((cmd, idx) => {
            const Comp = props.currentIdx === idx ? ListItemActive : ListItem;
            return (
                <Comp key={idx}>
                    {cmd.name}
                    {cmd.args.length > 0 && <span> - {cmd.args.join(', ')}</span>}
                </Comp>
            );
        })}
    </ul>
);

const StyledList = styled(CommandList)`
    margin: 0;
    padding: 0;
    list-style: none;
`;

export default StyledList;
