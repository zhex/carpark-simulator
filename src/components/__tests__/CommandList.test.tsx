import 'jest-styled-components';
import * as React from 'react';
import { create } from 'react-test-renderer';

import CommandList from '../CommandList';

const commands = [
    { name: 'PLACE', args: [0, 0, 'NORTH'] },
    { name: 'MOVE', args: [] },
    { name: 'RIGHT', args: [] },
    { name: 'REPORT', args: [] },
];

it('CommandList', () => {
    const tree = create(
        <CommandList commands={commands} currentIdx={2} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
