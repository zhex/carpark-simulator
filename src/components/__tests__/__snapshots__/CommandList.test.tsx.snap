// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandList Component 1`] = `
.c1 {
  padding: 10px;
  border-bottom: 1px solid #ccc;
}

.c1:last-of-type {
  border-bottom: none;
}

.c2 {
  padding: 10px;
  border-bottom: 1px solid #ccc;
  background: lightgreen;
}

.c2:last-of-type {
  border-bottom: none;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    PLACE
    <span>
       - 
      0, 0, NORTH
    </span>
  </li>
  <li
    className="c1"
  >
    MOVE
  </li>
  <li
    className="c2"
  >
    RIGHT
  </li>
  <li
    className="c1"
  >
    REPORT
  </li>
</ul>
`;
